packages/email/email.js

```diff
@@ -227,27 +225,66 @@ Email.customTransport = undefined;
+  Promise.await(Email.sendAsync(options));
+};
+
+/**
+ * @summary Send an email with asyncronous method. Capture  Throws an `Error` on failure to contact mail server
+ * or if mail server returns an error. All fields should match
+ * [RFC5322](http://tools.ietf.org/html/rfc5322) specification.
+ *
+ * If the `MAIL_URL` environment variable is set, actually sends the email.
+ * Otherwise, prints the contents of the email to standard out.
+ *
+ * Note that this package is based on **nodemailer**, so make sure to refer to
+ * [the documentation](http://nodemailer.com/)
+ * when using the `attachments` or `mailComposer` options.
+ *
+ * @locus Server
+ * @return {Promise}
+ * @param {Object} options
+ * @param {String} [options.from] "From:" address (required)
+ * @param {String|String[]} options.to,cc,bcc,replyTo
+ *   "To:", "Cc:", "Bcc:", and "Reply-To:" addresses
+ * @param {String} [options.inReplyTo] Message-ID this message is replying to
+ * @param {String|String[]} [options.references] Array (or space-separated string) of Message-IDs to refer to
+ * @param {String} [options.messageId] Message-ID for this message; otherwise, will be set to a random value
+ * @param {String} [options.subject]  "Subject:" line
+ * @param {String} [options.text|html] Mail body (in plain text and/or HTML)
+ * @param {String} [options.watchHtml] Mail body in HTML specific for Apple Watch
+ * @param {String} [options.icalEvent] iCalendar event attachment
+ * @param {Object} [options.headers] Dictionary of custom headers - e.g. `{ "header name": "header value" }`. To set an object under a header name, use `JSON.stringify` - e.g. `{ "header name": JSON.stringify({ tracking: { level: 'full' } }) }`.
+ * @param {Object[]} [options.attachments] Array of attachment objects, as
+ * described in the [nodemailer documentation](https://nodemailer.com/message/attachments/).
+ * @param {MailComposer} [options.mailComposer] A [MailComposer](https://nodemailer.com/extras/mailcomposer/#e-mail-message-fields)
+ * @param {Object} [options.stream] Output stream to write email on development environment
+ * object representing the message to be sent.  Overrides all other options.
+ * You can create a `MailComposer` object via
+ * `new EmailInternals.NpmModules.mailcomposer.module`.
+ */
+Email.sendAsync = async function (options) {
+  const { stream = process.stdout, ...rest } = options;
+  const email = rest.mailComposer ? rest.mailComposer.mail : rest;
 
   let send = true;
-  sendHooks.forEach(hook => {
-    send = hook(options);
+  sendHooks.forEach((hook) => {
+    send = hook(email);
     return send;
   });
-  if (!send) return;
+  if (!send) {
+    return;
+  }
 
-  const customTransport = Email.customTransport;
-  if (customTransport) {
+  if (Email.customTransport) {
     const packageSettings = Meteor.settings.packages?.email || {};
-    customTransport({ packageSettings, ...options });
-    return;
+    return Email.customTransport({ packageSettings, ...email });
```

Comment:
It's important to document, that if `Email.customTransport` returns a `Promise`, it'll be part of the `sendAsync` (and thus `send`) flow, i.e., if it takes a long time to process, it'll be included as well (previously it was called but not awaited).

Reply:
I changed it in the send synchronous method to preserve the current behavior.
Please see the changes here https://github.com/meteor/meteor/pull/12101/files#diff-81b3d1c2cbdddcb39f637030fdb2c2fe5e3190c2b3e0b638c68199f793aa7e5aR233. Is it make sense for you?

Reply:
My point was different: because `Email.sendAsync` now does `return Email.customTransport(...)`, doing `await Email.sendAsync` will not only wait for the code of `Email.sendAsync` itself, but also the `Email.customTransport`, due to promise chaining. I think it's a good change, but it must be documented.

Besides that, you can't just return `Email.sendAsync`, as it'll return a promise (it was `undefined` before). And because this promise is not `.catch`ed anywhere, it'll raise an unhandled promise rejection on error.